eth/src/main/kotlin/com/d3/eth/sidechain/EthChainListener.kt

```diff
@@ -16,6 +16,7 @@ import org.web3j.protocol.Web3j
 import org.web3j.protocol.core.DefaultBlockParameter
 import org.web3j.protocol.core.methods.response.EthBlock
 import java.math.BigInteger
+import java.util.concurrent.atomic.AtomicReference
```

Comment:
what is the purpose of atomic here since operations used are not atomic 😢

Reply:
It was just a suggestion. Please, don't pay attention till PR is ready for review.